x-pack/plugins/spaces/index.ts

```diff
@@ -116,69 +113,52 @@ export const spaces = (kibana: Record<string, any>) =>
-        await createDefaultSpace(server);
-      });
-
-      // Register a function that is called whenever the xpack info changes,
-      // to re-compute the license check results for this plugin.
-      xpackMainPlugin.info
-        .feature(thisPlugin.id)
-        .registerLicenseCheckResultsGenerator(checkLicense);
-
-      const spacesService = createSpacesService(server);
-      server.expose('getSpaceId', (request: any) => spacesService.getSpaceId(request));
-
-      const config = server.config();
-
-      const spacesAuditLogger = new SpacesAuditLogger(config, new AuditLogger(server, 'spaces'));
-
-      server.expose('spacesClient', {
-        getScopedClient: (request: Record<string, any>) => {
-          const adminCluster = server.plugins.elasticsearch.getCluster('admin');
-          const { callWithRequest, callWithInternalUser } = adminCluster;
-          const callCluster = (...args: any[]) => callWithRequest(request, ...args);
-          const { savedObjects } = server;
-          const internalRepository = savedObjects.getSavedObjectsRepository(callWithInternalUser);
-          const callWithRequestRepository = savedObjects.getSavedObjectsRepository(callCluster);
-          const authorization = server.plugins.security
-            ? server.plugins.security.authorization
-            : null;
-          return new SpacesClient(
-            spacesAuditLogger,
-            (message: string) => {
-              server.log(['spaces', 'debug'], message);
-            },
-            authorization,
-            callWithRequestRepository,
-            server.config(),
-            internalRepository,
-            request
-          );
+    async init(server: Server) {
+      const kbnServer = (server as unknown) as KbnServer;
+      const initializerContext = ({
+        legacyConfig: server.config(),
+        logger: {
+          get: (context: string) => ({
+            debug: (message: any) => server.log([context, 'debug'], message),
+            info: (message: any) => server.log([context, 'info'], message),
+            warning: (message: any) => server.log([context, 'warning'], message),
+            error: (message: any) => server.log([context, 'error'], message),
+          }),
+        },
+      } as unknown) as SpacesInitializerContext;
+
+      const spacesHttpService: SpacesHttpServiceSetup = {
+        ...(kbnServer.newPlatform.setup.core.http as HttpServerInfo),
+        route: server.route.bind(server),
+      };
+
+      const core: SpacesCoreSetup = {
+        http: spacesHttpService,
+        elasticsearch: kbnServer.newPlatform.setup.core.elasticsearch,
+        savedObjects: server.savedObjects,
+        usage: (server as any).usage,
+        tutorial: {
+          addScopedTutorialContextFactory: (server as any).addScopedTutorialContextFactory,
         },
+      };
+
+      const plugins = {
+        xpackMain: server.plugins.xpack_main,
+        getSecurity: () => server.plugins.security,
```

Comment:
Spaces has a circular dependency with Security right now. Security is not yet available when `init` runs, so this is wrapped in a function for the time being.

Reply:
can we add a comment in code?